storm-shim-9x/src/main/java/storm/mesos/shims/DockerCommandLineShim.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.mesos.shims;
+import java.util.Map;
+
+public class DockerCommandLineShim implements ICommandLineShim {
+  String extraConfig;
+
+  public DockerCommandLineShim(String extraConfig) {
+    this.extraConfig = extraConfig;
+  }
+
+  public String getCommandLine() {
+    // An ugly workaround for a bug in DCOS
```

Comment:
I think the idea was you'd be running the nimbus and the supervisor both in docker containers using the same image.  In that circumstance this avoids having to hardcode the path.  I don't know if I like it, as I might want to run nimbus w/o docker and but run the supervisors in docker but have different paths to libmesos.so.  I doubt I'd get much sympathy for that setup though.